packages/labs/analyzer/src/test/javascript/exports_test.ts

```diff
@@ -0,0 +1,263 @@
+/**
+ * @license
+ * Copyright 2022 Google LLC
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+import {suite} from 'uvu';
+// eslint-disable-next-line import/extensions
+import * as assert from 'uvu/assert';
+import {AbsolutePath} from '../../lib/paths.js';
+import {getSourceFilename, InMemoryAnalyzer, languages} from '../utils.js';
+
+for (const lang of languages) {
+  const test = suite<{
+    analyzer: InMemoryAnalyzer;
+  }>(`Exports tests (${lang})`);
+
+  test.before.each((ctx) => {
+    ctx.analyzer = new InMemoryAnalyzer(lang, {
+      '/package.json': JSON.stringify({name: '@lit-internal/in-memory-test'}),
+    });
+  });
+
```

Comment:
We don't have function analysis yet, so we'll add `export function` when we have that.

Otherwise, added support (and/or tests for):
* export {name as default}
* export default <expression>
* export * from 'module'
* export * as ns from 'module'